drivers/vcnl40x0/vcnl40x0_saul.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_vcnl40x0
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for VCNL40X0 device
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "vcnl40x0.h"
+#include "vcnl40x0_params.h"
+#include "xtimer.h"
+
+static int read_proximity(void *dev, phydat_t *res)
+{
+    vcnl40x0_t *d = (vcnl40x0_t *)dev;
+
+    res->val[0] = (int16_t)vcnl40x0_read_proximity(d);
+    res->unit = UNIT_M;
+    res->scale = 0;
```

Comment:
This suggests that the reading returned is in Meters, which seems rather bad resolution -> anything below 0.5m will return 0 and 0.5m - 1.0m will return 1m.